sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/TestMethodShouldHaveCorrectSignature.cs

```diff
@@ -44,83 +43,93 @@ public sealed class TestMethodShouldHaveCorrectSignature : SonarDiagnosticAnalyz
             DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(rule);
 
-        protected override void Initialize(SonarAnalysisContext context)
+        protected override void Initialize(SonarAnalysisContext context) =>
```

Comment:
I almost completely re-wrote the rule, so it's probably best not to try to compare it to the previous version.
There were two reasons for the re-write:
1) the previous version was looking at class declarations, then using _GetMethods_ on the class symbol to the methods.
2) I had to add support for another four or five types of test method attributes, some of which have different constraints on the signature e.g. Theories can be generic but Facts can't, so the validation checking got more complicated.